src/darwin/Framework/CHIP/MTRBaseDevice.mm

```diff
@@ -847,31 +847,37 @@ - (void)readAttributesWithEndpointID:(NSNumber * _Nullable)endpointID
     auto * bridge = new MTRDataValueDictionaryCallbackBridge(queue, completion,
         ^(ExchangeManager & exchangeManager, const SessionHandle & session, MTRDataValueDictionaryCallback successCb,
             MTRErrorCallback failureCb, MTRCallbackBridgeBase * bridge) {
+            // interactionStatus tracks whether the whole read interaction has failed.
+            //
+            // Make sure interactionStatus survives even if this block scope is destroyed.
+            auto interactionStatus = Platform::MakeShared<CHIP_ERROR>(CHIP_NO_ERROR);
+            if (interactionStatus == nullptr) {
```

Comment:
Is this checknecessary? I think generally in Darwin frameworks the assumption is that if a small malloc like this fails your process is about to crash anyway.

Reply:
I think the general convention is that Platform::* allocation can fail and we should check for that failure.  Realistically, if we don't care about that we could skip Platform and just std::make_shared here.

Reply:
Just switched to std::make_shared here.